### PR TITLE
[TrapFocus] Remove cast

### PR DESCRIPTION
**Pull request by AndrewMusgrave**
Causing production errors

`return element.querySelector(exports.FOCUSABLE_SELECTOR)`
`write(() => focusFirstFocusableNode(firstNode));`

### CODE REVIEW

File: src/components/TrapFocus/tests/TrapFocus.test.tsx [hunk 1]
```diff
@@ -1,17 +1,34 @@
...
 import {mountWithAppProvider, trigger} from 'test-utilities/legacy';
+import {mountWithApp} from 'test-utilities';
 import {
   EventListener,
   Focus,
   TextContainer,
   TextField,
   Button,
 } from 'components';
+import * as focusUtilities from '@shopify/javascript-utilities/focus';
```
Comment (AndrewMusgrave) [inline src/components/TrapFocus/tests/TrapFocus.test.tsx]: Imported like this to mock the value

File: src/components/TrapFocus/tests/TrapFocus.test.tsx [hunk 2]
```diff
@@ -1,17 +1,34 @@
...
 import {TrapFocus} from '../TrapFocus';
 
+jest.mock('@shopify/javascript-utilities/fastdom', () => ({
+  ...require.requireActual('@shopify/javascript-utilities/fastdom'),
+  write: (cb: () => void) => cb(),
```
Comment (AndrewMusgrave) [inline src/components/TrapFocus/tests/TrapFocus.test.tsx]: Mocking out RAF

File: src/components/TrapFocus/tests/TrapFocus.test.tsx [hunk 3]
```diff
@@ -1,17 +1,34 @@
...
+}));
+
 describe('<TrapFocus />', () => {
+  let focusFirstFocusableNodeSpy: jest.SpyInstance;
+
+  beforeEach(() => {
+    focusFirstFocusableNodeSpy = jest.spyOn(
```
Comment (AndrewMusgrave) [inline src/components/TrapFocus/tests/TrapFocus.test.tsx]: Spying on this so we can assert it's called with the correct node

File: UNRELEASED.md [hunk 4]
```diff
@@ -6,6 +6,8 @@
 
 ### Bug fixes
 
+- Fixed type-error in `TrapFocus` that causing `querySelector` being ran on null ([#2574](https://github.com/Shopify/polaris-react/pull/2574))
```
Comment (dleroux) [inline UNRELEASED.md]:
```suggestion
- Fixed type-error in `TrapFocus` that was causing `querySelector` being ran on null ([#2574](https://github.com/Shopify/polaris-react/pull/2574))
```

**Pull request merged**